### ISSUE TITLE
Corrupted metabase.db.mv.db file

### ISSUE THREAD

**Issue by playdribble** [metabase/metabase#3654]
Hi there,

Specs:
Metabase native Mac app v0.20.1
Running OS X Sierra 
Connecting to Mongo DB v3.0.12
Metabase internal database: H2 

I'm able to connect to the database, ask questions, and create dashboards with no problem initially. However, after having the installation for more than 24 hours, when opening or starting metabase, there is a "Fatal Error: The Metabase server terminated unexpectedly" error. This happens without making any changes to Metabase. I have also been able to reproduce this consistently a couple times.

I suspect that this has to do with the `metabase.db.mv.db` file being corrupted. When I delete that file from `~/Library/Application Support/Metabase`, I'm able to run Metabase again. However, my user profile and data is all gone. 

Please let me know if there's any other info I can provide for this issue. We had previously tried running this on an ec2 instance as well, with a similar problem. Does this potentially have to do with the MongoDB driver? Or potentially the metabase internal database (we've been using H2, might work if we switched to MySQL, etc).

Thanks for your help!

**Comment by salsakran** [comment]
Do you have the previous file?

Also, did you look into the Console application to see what the server logs had in them?

**Comment by playdribble** [comment]
I have the latest file, but not the file for when the metabase instance was working correctly.

Also, I did have a look into the console application. It only displayed connection/timeout errors (ie NSURLConnection error 1001, etc)

**Comment by salsakran** [comment]
can you share the corrupted file with us? We've gotten reports of this bug but it's been very difficult to reproduce.

**Comment by playdribble** [comment]
No problem, if you let me know the best way to get you the file, I'll send it over.

**Comment by salsakran** [comment]
could you email it to the address on my profile?

**Comment by Pin0** [comment]
I have the exact same issue:

Yesterday evening I was amazed with the result I got in a short time. This morning the window was gone, Metadata was running but after e restart I got the error.
Bummer I was ready to amaze my co workers...

Maybe it does not handle sleep of mac to well. I just folded my mac close.

Metabase version 0.20.3.0
Connecting to a postgres 9.3 remote database.

**Comment by camsaul** [comment]
@playdribble @Pin0 are you still seeing those issues with our latest release, 0.21.0? We've updated our H2 library which addressed a couple of problems with macOS Sierra and H2 in our latest release.

Either way, we highly recommend using [Postgres or MySQL as the application DB](http://www.metabase.com/docs/latest/operations-guide/start.html#configuring-the-metabase-application-database), *especially* if you're running it on an EC2 instance.

Please let me know if you're still seeing issues after you upgrade.

**Comment by bhanna** [comment]
Also having this issue and, like @Pin0 I just shut my mac to put it to sleep. Any updates?

**Comment by salsakran** [comment]
```
ERROR metabase.driver :: Failed to connect to database: org.h2.jdbc.JdbcSQLException: Database may be already in use: "Locked by another process: /Users/ben/Library/Application Support/Metabase/metabase.db.lock.db". Possible solutions: close all other connection(s); use the server mode [90020-193]
```

I think this is a separate issue. If you kill the process (and/or just restart your mac), the lock should be cleared. If you're getting a corruption error message, pretty please include as much info as you can re: versions of everything possible, as we can't reproduce this bug.

**Comment by bhanna** [comment]
Yes, restarting worked. If I get corruption I will snapshot everything and pass it along.

**Comment by salsakran** [comment]
glad to hear a restart worked, though it sucks you needed to do that.

**Comment by fandomfilms** [comment]
Getting fatal error when trying to load metabase

default	14:17:00.853790 -0500	Metabase	TIC TCP Conn Start [273:0x61800038f7d0]
default	14:17:00.857335 -0500	Metabase	TIC TCP Conn Event [273:0x61800038f7d0]: 3 Err(61)
error	14:17:00.857365 -0500	Metabase	TIC TCP Conn Failed [273:0x61800038f7d0]: 1:61 Err(61)
default	14:17:00.857380 -0500	Metabase	TIC TCP Conn Cancel [273:0x61800038f7d0]
default	14:17:00.857401 -0500	Metabase	TIC TCP Conn Destroyed [273:0x61800038f7d0]
error	14:17:00.857513 -0500	Metabase	HTTP load failed (error code: -1004 [1:61])
error	14:17:00.857597 -0500	Metabase	NSURLConnection finished with error - code -1004

**Comment by camsaul** [comment]
@fandomfilms looks like you posted this same comment in multiple issues. Check out my suggestions I wrote in response to your post in #4246

**Comment by camsaul** [comment]
As with other H2-related issues I believe this was resolved by #5396. Since we haven't heard anything about this issue since that got merged (and since we were able to find a workaround for this particular issue anyway) I'm going to close this out as fixed. @bhanna @ghost please feel free to open a new issue if you run into any further problems on an updated version of Metabase.
